Category.agda

```diff
@@ -0,0 +1,121 @@
+{-# OPTIONS --cubical #-}
+module Category where
+
+open import PathPrelude
+
+-- Functor
+record Func {ℓ} : Set (ℓ-suc ℓ) where
+  field
+    map      : Set ℓ → Set ℓ
+    fmap     : {A B : Set ℓ} → (A → B) → map A → map B
+    presId   : (A : Set ℓ) → fmap (idFun A) ≡ idFun (map A)
+    presComp : {A B C : Set ℓ} → (f : A → B) → (g : B → C) →
+                 fmap (g ∘ f) ≡ (fmap g) ∘ (fmap f)
+
+-- Functor Path
+record Func≡ {ℓ} : Set (ℓ-suc ℓ) where
+  field F G : Func {ℓ}
+  module L = Func F
+  module R = Func G
+
+  field
+    map≡      : (A : Set ℓ) →
+      Path
+      (L.map A)        (R.map A)
+    fmap≡     : {A B : Set ℓ} → (f : A → B) →
+      PathP (λ i → map≡ A i → map≡ B i)
+      (L.fmap f)       (R.fmap f)
+    presId≡   : (A : Set ℓ) →
+      PathP (λ i → fmap≡ (idFun A) i ≡ idFun (map≡ A i))
+      (L.presId A)     (R.presId A)
+    presComp≡ : {A B C : Set ℓ} → (f : A → B) → (g : B → C) →
+      PathP (λ i → fmap≡ (g ∘ f) i ≡ (fmap≡ g) i ∘ (fmap≡ f) i)
+      (L.presComp f g) (R.presComp f g)
+
+  path : F ≡ G
+  path i = record { map      = λ A   → map≡      A   i
+                  ; fmap     = λ f   → fmap≡     f   i
+                  ; presId   = λ A   → presId≡   A   i
+                  ; presComp = λ f g → presComp≡ f g i }
+
+idᶠ : ∀{ℓ} → Func {ℓ}
+idᶠ = record { map      = idFun (Set _)
+             ; fmap     = λ {A B} → idFun (A → B)
+             ; presId   = λ _ → refl
+             ; presComp = λ _ _ → refl }
+
+-- Composition of Functor
+infixr 45 _∘ᶠ_
+_∘ᶠ_ : ∀{ℓ} → Func {ℓ} → Func {ℓ} → Func {ℓ}
+G ∘ᶠ F = record
+  { map      = G.map ∘ F.map
+  ; fmap     = G.fmap ∘ F.fmap
+  ; presId   = λ A → trans (cong G.fmap (F.presId A)) (G.presId _)
+  ; presComp = λ f g → trans (cong G.fmap (F.presComp f g)) (G.presComp _ _)
+  } where module F = Func F
+          module G = Func G
+
+idᶠLeft : ∀{ℓ}{F : Func {ℓ}} → idᶠ ∘ᶠ F ≡ F
+idᶠLeft {ℓ} {F} = Func≡.path r where
+  module L = Func (idᶠ ∘ᶠ F)
+  module R = Func F
+  map≡ : L.map ≡ R.map
+  map≡ = funExt (λ _ → refl)
+  r = record { F         = idᶠ ∘ᶠ F
+             ; G         = F
+             ; map≡      = λ A → refl
+             ; fmap≡     = λ f → funExt λ _ → refl
+             ; presId≡   = λ A → {!!}
+             ; presComp≡ = λ f g → {!!}
+             }
```

Comment:
Indeed, this approach is similar to what I try before Func≡. However, it seem not to work.

Anyway I try to compile the following

`
idᶠLeft : ∀{ℓ}{F : Func {ℓ}} → idᶠ ∘ᶠ F ≡ F
idᶠLeft {ℓ} {F} i = record
  { map      = R.map
  ; fmap     = R.fmap
  ; presId   = {!!}
  ; presComp = {!!}
  } where module L = Func (idᶠ ∘ᶠ F)
          module R = Func F
`
and get this error message.

?0 : (A : Set ℓ) → R.fmap (idFun A) ≡ idFun (R.map A)
?1 : {A B C : Set ℓ} (f : A → B) (g : B → C) →
R.fmap ((λ {.x} → g) ∘ f) ≡ (λ {.x} → R.fmap g) ∘ R.fmap f

———— Errors ————————————————————————————————————————————————
Failed to solve the following constraints:
  [500] ?1 i1 f g i x = Func.presComp F f g i x : Func.map F C
  [499] ?0 i1 A i x = Func.presId F A i x : Func.map F A
  [491] ?1 i0 f g i x
        = primComp (λ _ → Func.map idᶠ (Func.map F C)) ((~ i) ∨ i)
          (λ z o →
             (λ { j .itIsOne → cong (Func.fmap idᶠ) (Func.presComp F f g) (~ j)
                ; j .itIsOne → Func.presComp idᶠ (Func.fmap F f) (Func.fmap F g) j
                ; .PathPrelude..extendedlambda0
                })
             z o
             (primComp (λ j → Func.map idᶠ (Func.map F A)) z
              (λ j → p[ x ] z (~ j)) x))
          (Func.fmap idᶠ ((λ {.x} → Func.fmap F g) ∘ Func.fmap F f)
           (primComp (λ j → Func.map idᶠ (Func.map F A)) i0
            (λ j → p[ x ] i0 (~ j)) x))
          : Func.map F C
  [490] ?0 i0 A i x
        = primComp (λ _ → Func.map idᶠ (Func.map F A)) ((~ i) ∨ i)
          (λ z o →
             (λ { j .itIsOne → cong (Func.fmap idᶠ) (Func.presId F A) (~ j)
                ; j .itIsOne → Func.presId idᶠ (Func.map F A) j
                ; .PathPrelude..extendedlambda0
                })
             z o
             (primComp (λ j → Func.map idᶠ (Func.map F A)) z
              (λ j → p[ x ] z (~ j)) x))
          (Func.fmap idᶠ (idFun (Func.map F A))
           (primComp (λ j → Func.map idᶠ (Func.map F A)) i0
            (λ j → p[ x ] i0 (~ j)) x))
          : Func.map F A